### DISCUSSION TITLE
Naming the workers

### DISCUSSION THREAD

**Discussion by NoReflexness** [pytest-dev/pytest-xdist#737] [General]
Is there a nice way to name the workers that are spun up when using --dist

As of now the workers are called gw0 and gw1 and so on. In my case these represent actual resources which has names and I would love to have these in the log output rather then:
![image](https://user-images.githubusercontent.com/2299330/144426088-0302bdfb-31a2-4968-a0a4-77e15428c252.png)

I have poked around to to try and identify where i should hack this in.. should I hack it in.. but have come up empty handed.

Can anyone shed light on how this might be possible or maybe give a lead to where I would have to look.

Thank in advance.

**Comment by RonnyPfannschmidt** [comment]
If I recall correctly, you can set up a Id in the tx

**Reply by NoReflexness** [reply]
wow thx for the quick response. I'll give that a go! just need to figure out what the 'tx' is. Any leads?

**Reply by NoReflexness** [reply]
I think i have found it.. ill play around with .. thanks again

**Reply by RonnyPfannschmidt** [reply]
sorry for leaving it out on mobile
this is all about https://execnet.readthedocs.io/en/latest/example/test_group.html#assigning-gateway-ids

**Reply by NoReflexness** [reply]
Yes perfect! Exactly what I needed. Top 🦄ness
